org.osgi.test.assertj.cm/src/main/java/org/osgi/test/assertj/cm/configuration/AbstractConfigurationAssert.java

```diff
@@ -0,0 +1,206 @@
+
+package org.osgi.test.assertj.cm.configuration;
+
+import java.util.Objects;
+
+import org.assertj.core.api.AbstractObjectAssert;
+import org.osgi.service.cm.Configuration;
+
+/**
+ * Abstract base class for {@link Configuration} specific assertions
+ */
+public abstract class AbstractConfigurationAssert<S extends AbstractConfigurationAssert<S, A>, A extends Configuration>
+	extends AbstractObjectAssert<S, A> {
+
+	/**
+	 * Creates a new <code>{@link AbstractConfigurationAssert}</code> to make
+	 * assertions on actual Configuration.
+	 *
+	 * @param actual the Configuration we want to make assertions on.
+	 */
+	protected AbstractConfigurationAssert(A actual, Class<S> selfType) {
+		super(actual, selfType);
+	}
+
+	/**
+	 * Verifies that the actual Configuration has no attributes.
+	 *
+	 * @return this assertion object.
+	 * @throws AssertionError if the actual Configuration's attributes is not
+	 *             empty.
+	 */
+	public S hasNoAttributes() {
+		// check that actual Configuration we want to make assertions on is not
+		// null.
+		isNotNull();
+
+		// we override the default error message with a more explicit one
+		String assertjErrorMessage = "\nExpecting :\n  <%s>\nnot to have attributes but had :\n  <%s>";
+
+		// check
+		if (actual.getAttributes()
+			.iterator()
+			.hasNext()) {
+			failWithMessage(assertjErrorMessage, actual, actual.getAttributes());
+		}
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual Configuration's bundleLocation is equal to the
+	 * given one.
+	 *
+	 * @param bundleLocation the given bundleLocation to compare the actual
+	 *            Configuration's bundleLocation to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual Configuration's bundleLocation is
+	 *             not equal to the given one.
+	 */
+	public S hasBundleLocation(String bundleLocation) {
+		// check that actual Configuration we want to make assertions on is not
+		// null.
+		isNotNull();
+
+		// overrides the default error message with a more explicit one
+		String assertjErrorMessage = "\nExpecting bundleLocation of:\n  <%s>\nto be:\n  <%s>\nbut was:\n  <%s>";
+
+		// null safe check
+		String actualBundleLocation = actual.getBundleLocation();
+		if (!Objects.deepEquals(actualBundleLocation, bundleLocation)) {
+			failWithMessage(assertjErrorMessage, actual, bundleLocation, actualBundleLocation);
+		}
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual Configuration's changeCount is equal to the
+	 * given one.
+	 *
+	 * @param changeCount the given changeCount to compare the actual
+	 *            Configuration's changeCount to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual Configuration's changeCount is not
+	 *             equal to the given one.
+	 */
+	public S hasChangeCount(long changeCount) {
```

Comment:
We probably want an assert that allows range testing like between, greater than, less than. Like IntegerAssert.

Reply:
will Address this in an extra PR with much more assert-methods and Conditions on the Configuration itself, here I would like to concentrate more on Configuration-events. I just took this as an `low hanging fruit` from the generator.